smartsim/_core/_cli/utils.py

```diff
@@ -68,28 +69,55 @@ def pip_install(
     """Install a pip package to be used in the SmartSim build
     Currently only Torch shared libraries are re-used for the build
     """
-    # form pip install command
-    cmd = ["python", "-m", "pip", "install"]
-    cmd.extend(packages)
-    if end_point:
-        cmd.extend(["-f", end_point])
+    packages.sort()
+    end_point_args = ("-f", str(end_point)) if end_point else ()
+
+    pkg_str_list = "\n\t".join(packages)
+    if verbose:
+        logger.info(f"Attempting to Install Packages:\n\t{pkg_str_list}")
+    for package in packages:
+        _pip("install", *end_point_args, package)
+    if verbose:
+        logger.info("Packages Installed Successfully")
 
-    cmd_arg = " ".join(cmd)
 
+def pip_uninstall(packages_to_remove: t.List[str], verbose: bool = False) -> None:
+    packages_to_remove.sort()
+    pkg_str_list = "\n\t".join(packages_to_remove)
     if verbose:
-        logger.info(f"Installing packages {packages}...")
-    # pylint: disable-next=consider-using-with
-    proc = subprocess.Popen(
-        cmd_arg, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
-    )
-    _, err = proc.communicate()
-    returncode = int(proc.returncode)
-    if returncode != 0:
-        error = f"{packages} installation failed with exitcode {returncode}\n"
-        error += err.decode("utf-8")
-        raise BuildError(error)
+        logger.info(f"Attempting to Uninstall Packages:\n\t{pkg_str_list}")
+    for package in packages_to_remove:
+        _pip("uninstall", "-y", package)
```

Comment:
could we avoid the hardcoded strings and make it easier to use by having:

```
def _pip(..., interactive=True):
   ...
```

this avoids yet another string constant to make sure case/format matches.

Reply:
change "-y" to kwarg